modules/carto/src/api/maps-v3-client.ts

```diff
@@ -268,32 +334,50 @@ async function _fetchMapDataset(dataset, accessToken, credentials, clientId) {
   });
 
   // Extract the last time the data changed
-  const cache = parseInt(new URL(url).searchParams.get('cache'), 10);
+  const cache = parseInt(new URL(url).searchParams.get('cache') || '', 10);
   if (cache && dataset.cache === cache) {
     return false;
   }
   dataset.cache = cache;
 
   // Only fetch if the data has changed
-  const data = await request({url, format: mapFormat, accessToken});
-  dataset.data = data;
+  dataset.data = await requestData({url, format: mapFormat, accessToken});
+
   return true;
 }
 
-async function fillInMapDatasets({datasets, token}, clientId, credentials) {
+async function fillInMapDatasets(
+  {datasets, token},
+  clientId: string,
+  credentials: CloudNativeCredentials
+) {
   const promises = datasets.map(dataset => _fetchMapDataset(dataset, token, credentials, clientId));
   return await Promise.all(promises);
 }
 
-export async function fetchMap({cartoMapId, clientId, credentials, autoRefresh, onNewData}) {
+export async function fetchMap({
+  cartoMapId,
+  clientId,
+  credentials,
+  autoRefresh,
+  onNewData
+}: {
+  cartoMapId: string;
+  clientId: string;
+  credentials?: CloudNativeCredentials;
+  autoRefresh?: number;
+  onNewData?: (map: any) => void;
+}) {
   const defaultCredentials = getDefaultCredentials();
   const localCreds = {
     ...(defaultCredentials.apiVersion === API_VERSIONS.V3 && defaultCredentials),
     ...credentials
-  };
+  } as CloudNativeCredentials;
   const {accessToken} = localCreds;
 
-  log.assert(cartoMapId, `Must define CARTO map id: fetchMap({cartoMapId: 'XXXX-XXXX-XXXX'})`);
+  // TODO: prettier conflicts with eslint here
+  // eslint-disable-next-line quotes
+  log.assert(cartoMapId, "Must define CARTO map id: fetchMap({cartoMapId: 'XXXX-XXXX-XXXX'})");
```

Comment:
As it is just an error message I would remove the single quotes around XXXX

Reply:
Hmm, i've changed to `"XXXX-XXXX-XXXX"` and both eslint and prettier are happy, i think it's good trade.